include/libsemigroups/forest.hpp

```diff
@@ -216,15 +226,26 @@ namespace libsemigroups {
     //! Constant
     // not noexcept because std::vector::operator[] isn't.
     [[nodiscard]] node_type parent(node_type i) const {
-      validate_node(i);
+      throw_if_node_out_of_bounds(i);
       return _parent[i];
     }
 
+    //! \brief Returns the parent of a node.
+    //!
+    //! \param i the node whose parent is sought.
+    //!
+    //! \returns
+    //! A \ref node_type.
+    //!
+    //! \complexity
+    //! Constant
+    //!
+    //! \warning No checks are performed on the arguments of this function.
```

Comment:
Same as with `operator!=`. The omission of a description after the brief description doesn't look quite right to me.